android-core/src/main/java/com/mparticle/internal/ConfigManager.java

```diff
@@ -277,7 +283,7 @@ void checkConfigStaleness() {
 
     @Nullable
     String getConfig() {
-        return sPreferences.getString(CONFIG_JSON, null);
+        return sPreferences.getString(CONFIG_JSON, "");
```

Comment:
I assume this change means that this will return a blank string instead of null if the config doesn't exist, but the method is marked nullable... Since this is an internal class, can we mark this non null since it will now always return a string?

Reply:
Ill review it, I think yes, but because its public I want to double-check it.

Reply:
if it's public, why is it inside a package called "internal"?

Reply:
MParticleOptions has always been in the internal package which in my opinion actually menas "core" (like where the core logic is). Most of the things, specifically in the Builder class needs to be public to actually be able to build the object.

Reply:
Gotcha, then yeah we can't change the API then, agreed.